mev-build-rs/src/auctioneer/service.rs

```diff
@@ -155,8 +155,13 @@ impl<
         context: Arc<Context>,
         genesis_time: u64,
     ) -> Self {
-        let relays =
-            parse_relay_endpoints(&config.relays).into_iter().map(Relay::from).collect::<Vec<_>>();
+        let relays = parse_relay_endpoints(&config.relays);
+        let mut relayset: HashSet<RelayEndpoint> = HashSet::new();
```

Comment:
you'd want `Ord` instead of `Eq`, but you could just use this: 

https://doc.rust-lang.org/std/vec/struct.Vec.html#method.dedup

or even skip the extra trait impls and just use: https://doc.rust-lang.org/std/vec/struct.Vec.html#method.dedup_by_key to point at the public key

note: the vec must be sorted, so you want to call `sort`, then `dedup` after getting the vec

I personally would prefer this to going through hashset given the data set will be so small

Reply:
gotcha, so use vec's dup of relayendpoint via ord on public key

Reply:
yep, which I think is already there!